addons/hw_drivers/controllers/driver.py

```diff
@@ -1,432 +1,325 @@
+    try:
+        path = os.getenv("HOME") + '/' + filename
+        f = open(path, 'r')
+        content = f.readline().strip('\n')
+        f.close()
+    finally:
+        return content
 
-    def get_connection(self):
-        pass
+def get_odoo_server_url():
+    return read_file_first_line('odoo-remote-server.conf')
 
-    def action(self, action):
-        pass
+def get_token():
+    return read_file_first_line('token')
 
 #----------------------------------------------------------
-# Usb drivers
+# Controllers
 #----------------------------------------------------------
-usbdrivers = []
-drivers = {}
-
-class UsbMetaClass(type):
-    def __new__(cls, clsname, bases, attrs):
-        newclass = super(UsbMetaClass, cls).__new__(cls, clsname, bases, attrs)
-        usbdrivers.append(newclass)
-        return newclass
-
-class USBDriver(Driver,metaclass=UsbMetaClass):
-    def __init__(self, dev):
-        super(USBDriver, self).__init__()
-        self.dev = dev
-        self.value = ""
-
-    def get_name(self):
-        lsusb = str(subprocess.check_output('lsusb')).split("\\n")
-        for usbpath in lsusb:  # Should filter on usb devices or inverse loops?
-            device = self.dev
-            if "%04x:%04x" % (device.idVendor, device.idProduct) in usbpath:
-                return usbpath.split("%04x:%04x" % (device.idVendor, device.idProduct))[1]
-        return str(device.idVendor) + ":" + str(device.idProduct)
-
-    def get_connection(self):
-        return 'direct'
-
-    def value(self):
-        return self.value
 
+class StatusController(http.Controller):
+    @http.route('/hw_drivers/action', type='json', auth='none', cors='*', csrf=False)
+    def action(self, device_id, data):
+        if device_id in iot_devices:
+            return iot_devices[device_id].action(data)
+        else:
+            return {'message': _('Device %s not found' % device_id)}
+
+    @http.route('/hw_drivers/event', type='json', auth='none', cors='*', csrf=False)
+    def event(self, requests):
+        req = DeviceManager.addRequest(requests)
+        if req['event'].wait(58):
+            req['event'].clear()
+            return_value = req['queue'].copy()
+            return return_value
+
+    @http.route('/hw_drivers/box/connect', type='http', auth='none', cors='*', csrf=False)
+    def connect_box(self, token):
+        server = get_odoo_server_url()
+        if server:
+            f = open('/var/www/False.jpg','rb')
+            return f.read()
+        else:
+            iotname = ''
+            token = b64decode(token).decode('utf-8')
+            url = token.split('|')[0]
+            token = token.split('|')[1]
+            reboot = 'noreboot'
+            path = os.getenv("HOME") + '/odoo/addons/point_of_sale/tools/posbox/configuration/connect_to_server.sh'
+            subprocess.call([path, url, iotname, token, reboot])
+            m.send_alldevices()
+            f = open('/var/www/True.jpg','rb')
+            return f.read()
```

Comment:
all of this certificate logic should move in a specific commit with the technical explanations + the why